### PR TITLE
Windowsizelock

### PR DESCRIPTION
**Pull request by Addaadostuff**
Its Functional, but the settings arent saved at this State

Menu items created,
Function linked
WindowManager extended

### CODE REVIEW

File: scenes/Mai7E04.tmp
Comment (quellus) [inline scenes/Mai7E04.tmp:1]: Not sure why Godot generates these tmp files, but we don't want them in the repo.

File: scenes/Mai9EBB.tmp
Comment (quellus) [inline scenes/Mai9EBB.tmp:1]: Delete this file too please

**Review by quellus — CHANGES_REQUESTED** [review]
There's a few things we need to do before we get it merged.

- [x] The spin boxes should automatically populate with the actual window size when the lock is enabled
- [x] The lock state as well as the resolution should be saved to and loaded from the save file

Thank you so much for contributing! Overall, I like this feature and I'm excited to see it added. On the small chance that you only wanted to work on it this far, let us know and we can finish it up. Otherwise, take your time and we'll keep the PR open until it's done :)

**Comment by Addaadostuff** [comment]
I can implement 

`The spin boxes should automatically populate with the actual window size when the lock is enabled`

but saving and loading is a problem right now.

Screenshot 2024-08-02 184645

Should not add it to version 0.9 and 0.10 wont work, and also 0.11 will be a proplem in loaddata 
`if version >= 0.11:`  means =>  0.2  > 0.11
I think the "Version" should change to string before adding these value in _load_data.

_save_data looks fine to already add it to.


What do you think?

**Review by Addaadostuff — COMMENTED** [review]
updated

**Comment by quellus** [comment]
Oof I forgot about the version number issue. We don't want to increment to 1.0 until we hit a milestone. Code and I were talking about changing version number to a string and we have it on the roadmap. Looks like we're at a tipping point where it needs to be our highest priority now. I think this issue might be blocked by that. I'll spend some time working on it today.

**Comment by Addaadostuff** [comment]
okay, 
if possible i would want to avoid making a rebase,  (bad memories from a lot of failed merges)
You could make another issue /pull request after you are done.
Or inlcude these 3 variables while you are already chancing the save/load functions.

**Comment by quellus** [comment]
I don't mind rebasing and resolving the merge conflicts! Whatever happens, I'll make sure your commits stay intact so you get credit for your contributions

**Comment by quellus** [comment]
I did the version string thing in #102 . Need Code's review though. There is a chance we won't get it merged until Sunday, but I'll poke him and see if he has time.

**Comment by quellus** [comment]
Code and I just looked at your latest changes in a call and here's our notes:
- Rename the check box to "Fixed window size"
- Make sure the Quit button is always the last button in the list
- Changing the numbers doesn't resize the window
- Every time the lock is toggled, the window shrinks by 10 pixels
- Code confirmed my suspicions that this is a really good feature and we're both really excited about it


We did the save system change. It shouldn't cause any merge conflicts. Just need to merge our main branch into yours!

**Comment by Addaadostuff** [comment]
> * Every time the lock is toggled, the window shrinks by 10 pixels

I stoped the shrinking, but i can't fix the margin changes on the title.

https://github.com/godotengine/godot/issues/81640

**Review by quellus — APPROVED** [review]
It gucci

**Review by code807 — APPROVED** [review]
Looks good! Thank you for the contribution!

**Pull request merged**
